unstructured/ingest/cli/cmds/sharepoint.py

```diff
@@ -55,6 +58,24 @@ def add_cli_options(cmd: click.Command) -> None:
                     https://[tenant]-admin.sharepoint.com.\
                     This requires the app to be registered at a tenant level",
             ),
+            click.Option(
+                ["--permissions-application-id"],
+                default=None,
+                type=str,
+                help="Application id for ingesting permission (rbac) data",
+            ),
+            click.Option(
+                ["--permissions-client-cred"],
+                default=None,
+                type=str,
+                help="Credentials for ingesting permission (rbac) data",
+            ),
+            click.Option(
+                ["--permissions-tenant"],
+                default=None,
+                type=str,
+                help="Sharepoint permission (rbac) tenant name, such as: abcde.onmicrosoft.com",
+            ),
```

Comment:
if one of these is specified is it required that the other two must also be specified? if so we should probably validate and throw the appropriate error

Reply:
Agreed, addressed with https://github.com/Unstructured-IO/unstructured/pull/1628/commits/e2fef65bf38ca763a72a658d325ec90e357b3bcb